components/atomic/dropdown/Dropdown/BaseDropdown.tsx

```diff
@@ -29,6 +29,26 @@ function BaseDropdown({ className, disclosure, submenu, label }: Props) {
     hidden: !active,
   };
 
+  const onOpenModal = {
+    onOpenModal: useCallback(
```

Comment:
It took me a minute to get that this is the buttonControlWithModal props. 😅 I'm seeing an error in the console:
```
Warning: Unknown event handler property `onOpenModal`. It will be ignored.
```
This is because the property `onOpenModal` is getting passed directly to the child element, and `onOpenModal` isn't a valid attribute on a button. It still works, just has an annoying error. Maybe it could be avoided with your refactor comment below.

(more info: https://reactjs.org/warnings/unknown-prop.html)

Reply:
@1aurend I think I figured it out! Instead of passing `onOpenModal` to the buttons below, why not set an `onClick` event to all the buttons that close the dropdown? Technically, we want the dropdown to close no matter which button is clicked. Since I'm not preventing the default click event from bubbling up, the modal opens as expected.
```
  const onClick = useCallback(() => {
    setActive(false);
  }, [setActive]);

  const submenuProps = {
    id: uid,
    "aria-hidden": !active,
    hidden: !active,
    onClick,
  };
```
You won't need to clone the elements again!

Reply:
@jen-castiron Thanks so much for all of this! And 🎊 for figuring out the custom click handler in `DialogueDisclosure`! I'm tempted to try adding the forward ref to `BaseDropDown` and to continue to pass the on `onClick` to each button in the list individually, probably changing the prop name to something like `closeDropdown` so it's clear what function it is. Then in `ButtonControl` we just always check for that prop and make sure to add it to the `onClick` if it's there? I think if add the forward ref and we're not cloning so many things, it will be a lot clearer what's going on in the dropdown and not so bad having to pass the close prop around to buttons?

Reply:
I'll also look into the unknown event handler issue. I think you're right that the refactor (and renaming it) will resolve that.